### ISSUE TITLE
$i18n type: t / te functions missing

### ISSUE THREAD

**Issue by Dominic-Marcelino** [nuxt-modules/i18n#1778]
### Environment

- Operating System: `Darwin`
- Node Version:     `v18.12.1`
- Nuxt Version:     `3.0.0`
- Runtime Modules:  `@nuxtjs/i18n@8.0.0-beta.7`, `@pinia/nuxt@0.4.6`

### Reproduction

````ts
const nuxtApp = useNuxtApp();
const { t } = nuxtApp.$i18n;
````

### Describe the bug

<img width="1058" alt="Bildschirmfoto 2022-12-31 um 12 52 30" src="https://user-images.githubusercontent.com/46492597/210135723-e7b48dfa-4a45-4e14-9ad6-6ba6ce8c2318.png">

When we get the translate function from the global nuxt-i18n instance we get a typescript warning as the function is not included in the types. Same for the te function (and I think even more, but haven't checked all)

**Comment by mrleblanc101** [comment]
Indeed, but it work even if the types are missing for those who are wondering.
Just faced this issue because I wanted to translate my confirm dialog prompt in Pinia.

**Comment by kazupon** [comment]
Sorry, my late response.
I had been sick while happy holidays...

If you have not still tried edge version, please check it!

**Comment by mrleblanc101** [comment]
@kazupon No worry, it not urgent as it's only a typescript issue so it can be bypassed pretty easily.

Trying right now.

**Comment by mrleblanc101** [comment]
@kazupon No worry, it not urgent as it's only a typescript issue so it can be bypassed pretty easily.

Trying right now.

**Comment by mrleblanc101** [comment]
@kazupon It works ! Also `@nuxtjs/i18n-edge` fix #1777

**Comment by VictorKrasnov** [comment]
I tried "@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge", but I have the same problem. My code example:

```
<template>
  <div>
    <test :title="$t('title')"/>
  </div>
</template>

<script setup lang="ts">
import Test from '~/components/test.vue'
</script>
```

It works, but IDE tell me: Unresolved function or method $t().

---

I also wrote like this:
```
<template>
  <div>
    <test :title="t('title')"/>
  </div>
</template>

<script setup lang="ts">
import Test from '~/components/test.vue'
const nuxtApp = useNuxtApp();
const { t } = nuxtApp.$i18n;
</script>
```

I got: 

TS2339: Property 't' does not exist on type 'ExportedGlobalComposer & NuxtI18nRoutingCustomProperties & I18nRoutingCustomProperties'.

---

My IDE is PHPStorm. In VSC I see the same notifications.

Is any way to fix it?

**Comment by kazupon** [comment]
Could you give us minimal reproduction github repo  or stackbliz, please?

In the case of Nuxt, the code is generated by nuxt itself and nuxtjs/i18n. Without your nuxt project environment, we can not work out if there is a problem.

**Comment by VictorKrasnov** [comment]
> Could you give us minimal reproduction github repo or stackbliz, please?

https://github.com/VictorKrasnov/i18n-demo

You can easy reproduce it yourself by next steps:

1. Run to create new Nuxt3 project: `npx nuxi init i18n-demo`
2. Add `"@nuxtjs/i18n": "npm:@nuxtjs/i18n-edge"` to **package.json**.
3. Edit **nuxt.config.ts** configuration (like in my code).
4. Edit **app.vue**. Add `{{ $t('title') }}` instead of NuxtWelcome.
5. Run `npm install`.
6. Run `npm run dev`.
7. Follow by address http://localhost:3000/, here you can see Russian version of app, add /en to URL to see English version. It's OK, it works fine.
8. But IDE shows error on $t() ... :(

**Comment by Dominic-Marcelino** [comment]
> If you have not still tried edge version, please check it!

Indeed the original issue is solved in the edge-version. Thanks!

Will leave the issue open in favour of @VictorKrasnov problem

**Comment by mrleblanc101** [comment]
Weird, it indeed doesn't work in @Dominic-Marcelino demo.
But in my app it work @kazupon.
After some debugging, I realized it work until I delete my `plugins/vueuse.ts` which is only import `import { vOnClickOutside } from '@vueuse/components';` and define it as a Global component.

I don't know how this could possibly be related, maybe `vue-demi` ?

![2023-01-09 12 17 36](https://user-images.githubusercontent.com/17392251/211368175-7244257d-1646-449d-b1f2-ea5e870f73b2.gif)

**Comment by kazupon** [comment]
@VictorKrasnov 
Thank you for your repro!
I've just faced your issue.

This issue is not related to nuxtjs/i18n, but also occurs in normal projects.
This issue is caused by a recent change in release flow in vue-i18n-next.
The `ComponentCustomProperties` extension `d.ts` should be npm published, but it seems to have been omitted.

I will now fix vue-i18n-next.

**Comment by kazupon** [comment]
I've just fixed and released edge-version, `8.0.0-beta.7-27888997.81e2956` 
please check it!

**Comment by VictorKrasnov** [comment]
Thank you so much! `npm update` solved the problem.
